pkg/network/server.go

```diff
@@ -715,29 +718,28 @@ func (s *Server) handleGetHeadersCmd(p Peer, gh *payload.GetBlockByIndex) error
 	return p.EnqueueP2PMessage(msg)
 }
 
-const extensibleVerifyMaxGAS = 2000000
-
 // handleExtensibleCmd processes received extensible payload.
 func (s *Server) handleExtensibleCmd(e *payload.Extensible) error {
-	if err := s.chain.VerifyWitness(e.Sender, e, &e.Witness, extensibleVerifyMaxGAS); err != nil {
+	ok, err := s.extensiblePool.Add(e)
+	if err != nil {
 		return err
 	}
-	h := s.chain.BlockHeight()
-	if h < e.ValidBlockStart || e.ValidBlockEnd <= h {
-		// We can receive consensus payload for the last or next block
-		// which leads to unwanted node disconnect.
-		if e.ValidBlockEnd == h {
-			return nil
-		}
-		return errors.New("invalid height")
+	if !ok { // payload is already in cache
+		return nil
 	}
-
 	switch e.Category {
 	case consensus.Category:
 		s.consensus.OnPayload(e)
 	default:
 		return errors.New("invalid category")
 	}
+
+	msg := NewMessage(CMDInv, payload.NewInventory(payload.ExtensibleType, []util.Uint256{e.Hash()}))
+	if e.Category == consensus.Category {
+		s.broadcastHPMessage(msg)
+	} else {
```

Comment:
`else` is impossible here because of an error 5 lines above.

Reply:
It's OK to still have it, we'll have more categories soon.